File: 2166-design-bitset/2166-design-bitset.java
class Bitset {
    StringBuilder s = new StringBuilder(), flipped = new StringBuilder();
    int siz = 0, ones = 0;
    public Bitset(int size) {
        siz = size;
        for (int i = 0; i < size; i++) {
            s.append(0); flipped.append(1);
        }
    }
    public void fix(int idx) {
        if(s.charAt(idx) == '0') ones++;  
        s.setCharAt(idx, '1'); flipped.setCharAt(idx, '0'); 
    }
    public void unfix(int idx) {
        if(s.charAt(idx) == '1') ones--;
        s.setCharAt(idx, '0'); flipped.setCharAt(idx, '1'); 
    }
    public void flip() {
        StringBuilder temp = s;
        s = flipped;
        flipped = temp;
        ones = siz - ones;
    }
    public boolean all() { return ones == siz; }
    public boolean one() { return ones > 0; }
    public int count() { return ones; }
    public String toString() { return s.toString(); }
}

File: 2166-design-bitset/2166-design-bitset.cpp
class Bitset {
public:
     int a = 0, sz = 0, cnt = 0, isFlip = 0;
    string s;
    Bitset(int size) {
        sz = size;
        s = string(sz, '0');
    }
    
    void fix(int idx) {
        if (s[idx] == '0' + isFlip) {
            s[idx] = '1' + '0' - s[idx];
            cnt++;
        }
    }
    
    void unfix(int idx) {
        if (s[idx] == '1' - isFlip) {
            s[idx] = '1' + '0' - s[idx];
            cnt--;
        }
    }
    
    void flip() {
        isFlip ^= 1;
        cnt = sz - cnt;
    }
    
    bool all() {
        return cnt == sz;
    }
    
    bool one() {
        return cnt > 0;
    }
    
    int count() {
        return cnt;
    }
    
    string toString() {
        if (isFlip) {
            string s2 = s;
            for (auto& c: s2)
                c = '0' + '1' - c;
            return s2;
        }
        return s;
    }
};

